README.md

```diff
@@ -1,2 +1,307 @@
-# buildingimages
+# Building Container Images in IBM Cloud
+
 content creation around image building for containers
+
+# Prerequisites
+
+Before you get started you'll need an IBM Cloud account and need to
+install
+the
+[IBM Cloud Development Tools](https://github.com/IBM-Cloud/ibm-cloud-developer-tools).
+
+You should also clone
+the [guestbook repository](https://github.com/IBM/guestbook) as all
+image builds will be based on that.
+
+Once those are installed, you'll be able to complete the commands
+below.
+
+# Basics of Image Building
+
+Kubernetes uses standard container images to deploy applications.
+
+From the application's perspective, a container image is a single
+filesystem that is created before the application is started.
+
+From a creation perspective, container images are layers. These layers
+are built up using a Docker file. Lets look at the example guestbook
+application Docker file and pull it apart.
+
+```
+FROM golang as builder
```

Comment:
This is actually a fairly advanced Dockerfile given it does more than one build - and as such we need to either 1) split it into two for education purposes, or 2) add lots of text explaining what's going on.  Either is ok with me, as long as the reader isn't lost.

Reply:
ACK. I'm happy replacing the image build with something that uses a more standard linux base. I've got some examples in the ny-power side of containerizing the mosquitto service.

Reply:
I'm not concerned about the base image, its that we're doing 2 builds in one Dockerfile. That's a relatively new feature of docker build and not a newbie concept.  Thinking about it more.... I wouldn't mind if the doc for this just showed what the base build Dockerfile would look like even though its not in the repo, meaning:
```
+FROM golang
+RUN go get github.com/codegangsta/negroni
+RUN go get github.com/gorilla/mux github.com/xyproto/simpleredis
+COPY main.go .
+RUN go build main.go
```
And talked about these bits, then said something like "this is fine but what we really want is another image after that that doesn't have the go compiler, it just has our exe". Then introduce the 2nd half.  Then shows the optimization of doing it all in one Dockerfile. This will then also hammer home your idea of "cleaning up stuff as you build". Doing so isn't always easy to do, like having the go compiler as part of the base image.

Of course you could have the 2 Dockefriles as part of "this" repo instead of guestbook, and just point to guestbook's as the final (optimized) result.